### PR TITLE
Add validation for service with denied pool by constraints

### PR DESCRIPTION
**Pull request by wpjunior**

### CODE REVIEW

File: provision/pool/constraint.go [hunk 1]
```diff
@@ -263,3 +266,33 @@ func ToConstraintType(value string) (poolConstraintType, error) {
...
+	poolAllowedServices, err := p.GetServices()
+	if err != nil {
+		return err
+	}
+
+	if !sliceContains(poolAllowedServices, service) {
+		return &tsuruErrors.HTTP{
+			Code:    http.StatusBadRequest,
+			Message: fmt.Sprintf("Service %q is not available in pool %q", service, pool),
+		}
```
Comment (nettoclaudio) [inline provision/pool/constraint.go:285]: WDYT returning an error message with the available services as well? Pretty much like we do in routers validation (see [`ValidateRouters`](https://github.com/tsuru/tsuru/blob/main/provision/pool/pool.go#L247))

**Review by nettoclaudio — CHANGES_REQUESTED** [review]
:zap:

File: api/service_instance.go [hunk 2]
```diff
@@ -86,6 +86,14 @@ func createServiceInstance(w http.ResponseWriter, r *http.Request, t auth.Token)
 			return permission.ErrUnauthorized
 		}
 	}
+
+	if srv.IsMultiCluster && instance.Pool != "" {
+		err = pool.CheckServiceAllowed(ctx, instance.Pool, instance.ServiceName)
+		if err != nil {
+			return err
+		}
+	}
```
[the remark this answers is not in the document]
Comment (wpjunior) [reply api/service_instance.go]: Done @nettoclaudio, thanks for you suggestion

**Review by tcarreira — APPROVED** [review]

**Review by nettoclaudio — APPROVED** [review]
Pretty good refactor. :tada:

**Pull request merged**